Add conversation streaming (#409)
xmtp/docs-xmtp-org: docs/pages/agents/build-agents/stream.mdx

:::

## Stream new group chat and DM conversations

> With the agent-sdk, you can listen for new conversations using event handlers. The agent automatically handles streaming and provides simple event-based APIs.

Reviewer: ```suggestion
With the Agent SDK, you can listen for new conversations using event handlers. The agent automatically handles streaming and provides simple event-based APIs.
```

> ```tsx

Reviewer: ```suggestion
```ts
```

// Listen for new DM conversations
agent.on("dm", async (ctx) => {
  console.log("New DM conversation:", ctx.conversation.id);
});
// Listen for new group conversations
agent.on("group", async (ctx) => {
  console.log("New group conversation:", ctx.conversation.id);
});
``` 

##  Handle unhandled messages

> The `unhandledMessage` event is triggered for messages that don't match any of the support content types.

Reviewer: ```suggestion
The `unhandledMessage` event is triggered for messages that don't match any of the [default content types](https://docs.xmtp.org/agents/content-types/content-types#standards-track-content-types).
```
